### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by JonDouglas**
Creates an initial CONTRIBUTING.md file based on https://github.com/NuGet/Home/wiki/Submitting-Bugs-and-Suggestions and https://github.com/NuGet/Home/wiki/Contribute-to-NuGet

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+If you cannot find an existing issue that describes your bug or feature, submit an issue using the guidelines below.
+
+## Bug Reports
+A bug is a demonstrable problem that is caused by code in the repository. Good bug reports are extremely helpful for our developers who maintain this project.
+
+Guidelines for bug reports:
+
+1. Validate your code - Ensure that your problem isn’t caused by an error in your code.
+1. Use the GitHub issue search - Check if the issue has already been reported.
```
Comment (zivkan) [inline CONTRIBUTING.md:24]: Personally, I find GitHub's issue search of very limited use. Mainly because it sorts issues by creation date, rather than relevance, which is how basically every other search in any other product works.

I find using a web search engine with "my keywords site:github.com/NuGet/Home" to be the most effective way to search for existing issues. Should we add such a recommendation?

**Review by zivkan — APPROVED** [review]

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+Suggestions and feature requests are welcomed. Before opening a suggestion or feature request, please take a moment to find out whether your idea fits with the scope and goals of the project. It’s up to you to make a strong case to convince the project maintainers and community the merits of the suggestion or feature. Please provide as much detail and context as possible.
+
+Additionally, do check the existing [Issues](https://github.com/NuGet/Home/issues), [Pull Requests](https://github.com/NuGet/Home/pulls), and [NuGet Designs](https://github.com/NuGet/Home/tree/dev/designs) to see if your idea is already captured. If it is, consider adding feedback to the existing issue, pull request, or contribute directly to the design.
+
+## Pull requests
+Good pull requests of patches, improvements, new features, etc are great contributions. They should however remain focused in scope and avoid unrelated commits.
+
+Before embarking on any significant pull request, please ask a maintainer if the scope of the pull request would be reasonable to merge into the project. Otherwise you can potentially waste a lot of time working on something that might not be relevant to the project’s goals.
```
Comment (joelverhagen) [inline CONTRIBUTING.md]: Should we mention that major (or most? or all?) PRs should only come after an issue has been filed and there's some agreement?

The `please ask a maintainer` part should be clear so people know HOW to ask.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+If you find your issue already exists, make relevant comments and add your [reaction](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments). Use a reaction in place of a "+1" comment.
+
+👍 - upvote
+
+👎 - downvote
+
+The NuGet project is distributed across multiple repositories, try to file the issue against the correct repository [Related Projects](https://github.com/NuGet/Home#repos-and-projects).
+
+If your issue is a question then please ask the question on [Stack Overflow](https://stackoverflow.com/questions/tagged/nuget) using the tag `nuget`. Additionally, you can ask a question using [Microsoft Q&A](https://docs.microsoft.com/en-us/answers/topics/dotnet-package-management.html) under the `package management` section.
```
Comment (joelverhagen) [inline CONTRIBUTING.md:14]: Is the `nuget` tag on Stack Overflow... good? Like is this an effective way for a struggling user to actually get help? I'm curious if we have some idea of what the user experience is in this regard. I looked at the [`nuget`](https://stackoverflow.com/questions/tagged/nuget) tag on StackOverflow and see mostly unanswered questions.

**Review by joelverhagen — APPROVED** [review]
Good stuff. I think this is definitely good enough to ship as is. It's easy to tune this as we go.
Nice work!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+## Pull requests
+Good pull requests of patches, improvements, new features, etc are great contributions. They should however remain focused in scope and avoid unrelated commits.
+
+Before embarking on any significant pull request, please ask a maintainer if the scope of the pull request would be reasonable to merge into the project. Otherwise you can potentially waste a lot of time working on something that might not be relevant to the project’s goals.
+
+While creating a pull request, please also include the intent for which you'd like the changes to land. By default, we ask that you pull against the `dev` branch. However if you'd like to see a change in the next service release or preview release of Visual Studio, please communicate with a maintainer of this project to understand if the changes will be able to make the intended release.
```
Comment (nkolev92) [inline CONTRIBUTING.md:47]: Should this section point to https://github.com/NuGet/NuGet.Client/blob/dev/CONTRIBUTING.md in some shape or form?

File: CONTRIBUTING.md (hunk 3 above)
Comment (JonDouglas) [reply CONTRIBUTING.md:14]: It's a very low traffic tag, but it is one method to be used alongside the Microsoft Q&A, GitHub, etc. We should have more questions on SO imo!

image

File: CONTRIBUTING.md (hunk 4 above)
Comment (JonDouglas) [reply CONTRIBUTING.md:47]: At the bottom I link to https://github.com/NuGet/Home/wiki/Contribute-to-NuGet which then links to https://github.com/NuGet/NuGet.Client/blob/dev/CONTRIBUTING.md

Comment (JonDouglas) [reply CONTRIBUTING.md:47]: We should probably repurpose the Wiki page into this doc eventually?

File: CONTRIBUTING.md (hunk 2 above)
Comment (JonDouglas) [reply CONTRIBUTING.md]: Added a sentence to fix this :)

File: CONTRIBUTING.md (hunk 4 above)
Comment (nkolev92) [reply CONTRIBUTING.md:47]: Yeah, we can do that...the dotnet wide trend seems to be using the repo content for info like this.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+## Pull requests
+Good pull requests of patches, improvements, new features, etc are great contributions. They should however remain focused in scope and avoid unrelated commits.
+
+Before embarking on any significant pull request, please ask a maintainer in a GitHub issue if the scope of the pull request would be reasonable to merge into the project. Otherwise you can potentially waste a lot of time working on something that might not be relevant to the project’s goals.
+
+While creating a pull request, please also include the intent for which you'd like the changes to land. By default, we ask that you pull against the `dev` branch. However if you'd like to see a change in the next service release or preview release of Visual Studio, please communicate with a maintainer of this project to understand if the changes will be able to make the intended release.
+
+## Contributing Fixes
+If you are interested in fixing issues and contributing directly to the code base, please see the document [[Contribute To NuGet|Contribute-to-NuGet]].
```
Comment (loic-sharma) [inline CONTRIBUTING.md]: Does this syntax work? The preview didn't turn this into a link

**Review by loic-sharma — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (JonDouglas) [reply CONTRIBUTING.md:24]: I don't think we need to go too far into the recommendations here. I can also reword it to say "Search GitHub Issues - Check if the issue has already been reported"

File: CONTRIBUTING.md (hunk 5 above)
Comment (JonDouglas) [reply CONTRIBUTING.md]: Fixed with a link instead! 👍

**Comment by JonDouglas** [comment]
Last call for feedback! Merging in the morning!

File: CONTRIBUTING.md (hunk 4 above)
Comment (Soffieswan69) [reply CONTRIBUTING.md:47]: Duplicate of #

File: CONTRIBUTING.md (hunk 1 above)
Comment (Soffieswan69) [reply CONTRIBUTING.md:24]: Duplicate of #

File: CONTRIBUTING.md (hunk 3 above)
Comment (Soffieswan69) [reply CONTRIBUTING.md:14]: contributing

**Review by Soffieswan69 — COMMENTED** [review]
Duplicate of #
